### PR TITLE
Fix AUTO_INCREMENT for complex primary keys

### PR DESCRIPTION
**Pull request by loryk**
### Description

In case we are use complex primary keys, one or many of them can be auto incremented fields. So I think need to add AUTO_INCREMENT without check PRIMIRY is simple or complex.

### CODE REVIEW

**Comment by slnode** [comment]
Can one of the admins verify this patch? To accept patch and trigger a build add comment ".*ok\W+to\W+test.*"

**Comment by loryk** [comment]
Any news?

File: test/migration.test.js [hunk 1]
```diff
@@ -10,7 +10,7 @@ var platform = require('./helpers/platform');
 var should = require('./init');
 var Schema = require('loopback-datasource-juggler').Schema;
 
-var db, UserData, StringData, NumberData, DateData, DefaultData, SimpleEmployee;
+var db, UserData, ComlexData, StringData, NumberData, DateData, DefaultData, SimpleEmployee;
```
Comment (dhmlau) [inline test/migration.test.js]: For `ComlexData`, do you mean `ComplexData`?

**Comment by dhmlau** [comment]
@slnode test please

**Comment by dhmlau** [comment]
@loryk, there are 2 test cases failed for Node.js 12.  Could you please take a look? 

```
22:05:36 
22:05:36   1) MySQL specific datatypes Support explicit datatypes on a property find an instance:
22:05:36 
22:05:36       Uncaught AssertionError [ERR_ASSERTION]: The expression evaluated to a falsy value:
22:05:36 
22:05:36   assert(_.isEqual(data[0].amount, result[0].amount))
22:05:36 
22:05:36       + expected - actual
22:05:36 
22:05:36       -false
22:05:36       +true
22:05:36       
22:05:36       at test/datatypes.test.js:93:9
22:05:36       at node_modules/loopback-datasource-juggler/lib/dao.js:1709:9
22:05:36       at node_modules/async/dist/async.js:1140:9
22:05:36       at node_modules/async/dist/async.js:473:16
22:05:36       at iteratorCallback (node_modules/async/dist/async.js:1064:13)
22:05:36       at node_modules/async/dist/async.js:969:16
22:05:36       at node_modules/async/dist/async.js:1137:13
22:05:36       at buildResult (node_modules/loopback-datasource-juggler/lib/dao.js:1675:11)
22:05:36       at node_modules/loopback-datasource-juggler/lib/dao.js:1689:13
22:05:36       at doNotify (node_modules/loopback-datasource-juggler/lib/observer.js:157:49)
22:05:36       at doNotify (node_modules/loopback-datasource-juggler/lib/observer.js:157:49)
22:05:36       at Function.ObserverMixin._notifyBaseObservers (node_modules/loopback-datasource-juggler/lib/observer.js:180:5)
22:05:36       at Function.ObserverMixin.notifyObserversOf (node_modules/loopback-datasource-juggler/lib/observer.js:155:8)
22:05:36       at Function.ObserverMixin._notifyBaseObservers (node_modules/loopback-datasource-juggler/lib/observer.js:178:15)
22:05:36       at Function.ObserverMixin.notifyObserversOf (node_modules/loopback-datasource-juggler/lib/observer.js:155:8)
22:05:36       at withNotify (node_modules/loopback-datasource-juggler/lib/dao.js:1687:17)
22:05:36       at node_modules/loopback-datasource-juggler/lib/dao.js:1626:11
22:05:36       at node_modules/async/dist/async.js:1135:9
22:05:36       at eachOfArrayLike (node_modules/async/dist/async.js:1069:9)
22:05:36       at eachOf (node_modules/async/dist/async.js:1117:5)
22:05:36       at _asyncMap (node_modules/async/dist/async.js:1133:5)
22:05:36       at Object.map (node_modules/async/dist/async.js:1122:16)
22:05:36       at allCb (node_modules/loopback-datasource-juggler/lib/dao.js:1621:13)
22:05:36       at node_modules/loopback-connector/lib/sql.js:1472:7
22:05:36       at node_modules/loopback-datasource-juggler/lib/observer.js:269:22
22:05:36       at doNotify (node_modules/loopback-datasource-juggler/lib/observer.js:157:49)
22:05:36       at MySQL.ObserverMixin._notifyBaseObservers (node_modules/loopback-datasource-juggler/lib/observer.js:180:5)
22:05:36       at MySQL.ObserverMixin.notifyObserversOf (node_modules/loopback-datasource-juggler/lib/observer.js:155:8)
22:05:36       at cbForWork (node_modules/loopback-datasource-juggler/lib/observer.js:259:14)
22:05:36       at node_modules/loopback-connector/lib/sql.js:661:9
22:05:36       at handleResponse (lib/mysql.js:193:17)
22:05:36       at Query.<anonymous> (lib/mysql.js:204:7)
22:05:36       at Query.<anonymous> (node_modules/mysql/lib/Connection.js:525:10)
22:05:36       at Query._callback (node_modules/mysql/lib/Connection.js:491:16)
22:05:36       at Query.Sequence.end (node_modules/mysql/lib/protocol/sequences/Sequence.js:83:24)
22:05:36       at Query._handleFinalResultPacket (node_modules/mysql/lib/protocol/sequences/Query.js:139:8)
22:05:36       at Query.EofPacket (node_modules/mysql/lib/protocol/sequences/Query.js:123:8)
22:05:36       at Protocol._parsePacket (node_modules/mysql/lib/protocol/Protocol.js:291:23)
22:05:36       at Parser._parsePacket (node_modules/mysql/lib/protocol/Parser.js:433:10)
22:05:36       at Parser.write (node_modules/mysql/lib/protocol/Parser.js:43:10)
22:05:36       at Protocol.write (node_modules/mysql/lib/protocol/Protocol.js:38:16)
22:05:36       at Socket.<anonymous> (node_modules/mysql/lib/Connection.js:91:28)
22:05:36       at Socket.<anonymous> (node_modules/mysql/lib/Connection.js:525:10)
22:05:36       at addChunk (_stream_readable.js:305:12)
22:05:36       at readableAddChunk (_stream_readable.js:286:11)
22:05:36       at Socket.Readable.push (_stream_readable.js:220:10)
22:05:36       at TCP.onStreamRead (internal/stream_base_commons.js:166:17)
22:05:36 
22:05:36   2) MySQL specific datatypes Support explicit datatypes on a property find an instance by id:
22:05:36 
22:05:36       Uncaught AssertionError [ERR_ASSERTION]: The expression evaluated to a falsy value:
22:05:36 
22:05:36   assert(_.isEqual(data[0].amount, result.amount))
22:05:36 
22:05:36       + expected - actual
22:05:36 
22:05:36       -false
22:05:36       +true
22:05:36       
22:05:36       at test/datatypes.test.js:105:9
22:05:36       at node_modules/loopback-datasource-juggler/lib/dao.js:1774:5
22:05:36       at node_modules/loopback-datasource-juggler/lib/dao.js:1709:9
22:05:36       at node_modules/async/dist/async.js:1140:9
22:05:36       at node_modules/async/dist/async.js:473:16
22:05:36       at iteratorCallback (node_modules/async/dist/async.js:1064:13)
22:05:36       at node_modules/async/dist/async.js:969:16
22:05:36       at node_modules/async/dist/async.js:1137:13
22:05:36       at buildResult (node_modules/loopback-datasource-juggler/lib/dao.js:1675:11)
22:05:36       at node_modules/loopback-datasource-juggler/lib/dao.js:1689:13
22:05:36       at doNotify (node_modules/loopback-datasource-juggler/lib/observer.js:157:49)
22:05:36       at doNotify (node_modules/loopback-datasource-juggler/lib/observer.js:157:49)
22:05:36       at Function.ObserverMixin._notifyBaseObservers (node_modules/loopback-datasource-juggler/lib/observer.js:180:5)
22:05:36       at Function.ObserverMixin.notifyObserversOf (node_modules/loopback-datasource-juggler/lib/observer.js:155:8)
22:05:36       at Function.ObserverMixin._notifyBaseObservers (node_modules/loopback-datasource-juggler/lib/observer.js:178:15)
22:05:36       at Function.ObserverMixin.notifyObserversOf (node_modules/loopback-datasource-juggler/lib/observer.js:155:8)
22:05:36       at withNotify (node_modules/loopback-datasource-juggler/lib/dao.js:1687:17)
22:05:36       at node_modules/loopback-datasource-juggler/lib/dao.js:1626:11
22:05:36       at node_modules/async/dist/async.js:1135:9
22:05:36       at eachOfArrayLike (node_modules/async/dist/async.js:1069:9)
22:05:36       at eachOf (node_modules/async/dist/async.js:1117:5)
22:05:36       at _asyncMap (node_modules/async/dist/async.js:1133:5)
22:05:36       at Object.map (node_modules/async/dist/async.js:1122:16)
22:05:36       at allCb (node_modules/loopback-datasource-juggler/lib/dao.js:1621:13)
22:05:36       at node_modules/loopback-connector/lib/sql.js:1472:7
22:05:36       at node_modules/loopback-datasource-juggler/lib/observer.js:269:22
22:05:36       at doNotify (node_modules/loopback-datasource-juggler/lib/observer.js:157:49)
22:05:36       at MySQL.ObserverMixin._notifyBaseObservers (node_modules/loopback-datasource-juggler/lib/observer.js:180:5)
22:05:36       at MySQL.ObserverMixin.notifyObserversOf (node_modules/loopback-datasource-juggler/lib/observer.js:155:8)
22:05:36       at cbForWork (node_modules/loopback-datasource-juggler/lib/observer.js:259:14)
22:05:36       at node_modules/loopback-connector/lib/sql.js:661:9
22:05:36       at handleResponse (lib/mysql.js:193:17)
22:05:36       at Query.<anonymous> (lib/mysql.js:204:7)
22:05:36       at Query.<anonymous> (node_modules/mysql/lib/Connection.js:525:10)
22:05:36       at Query._callback (node_modules/mysql/lib/Connection.js:491:16)
22:05:36       at Query.Sequence.end (node_modules/mysql/lib/protocol/sequences/Sequence.js:83:24)
22:05:36       at Query._handleFinalResultPacket (node_modules/mysql/lib/protocol/sequences/Query.js:139:8)
22:05:36       at Query.EofPacket (node_modules/mysql/lib/protocol/sequences/Query.js:123:8)
22:05:36       at Protocol._parsePacket (node_modules/mysql/lib/protocol/Protocol.js:291:23)
22:05:36       at Parser._parsePacket (node_modules/mysql/lib/protocol/Parser.js:433:10)
22:05:36       at Parser.write (node_modules/mysql/lib/protocol/Parser.js:43:10)
22:05:36       at Protocol.write (node_modules/mysql/lib/protocol/Protocol.js:38:16)
22:05:36       at Socket.<anonymous> (node_modules/mysql/lib/Connection.js:91:28)
22:05:36       at Socket.<anonymous> (node_modules/mysql/lib/Connection.js:525:10)
22:05:36       at addChunk (_stream_readable.js:305:12)
22:05:36       at readableAddChunk (_stream_readable.js:286:11)
22:05:36       at Socket.Readable.push (_stream_readable.js:220:10)
22:05:36       at TCP.onStreamRead (internal/stream_base_commons.js:166:17)
22:05:36 
22:05:36 
```

**Comment by jannyHou** [comment]
The two failures should be fixed with the latest merged PR.

**Comment by jannyHou** [comment]
@slnode test please

**Comment by dhmlau** [comment]
@loryk, I've rebased your branch and fixed the typo as well.  Let's wait for CI to finish. Thanks.

**Comment by dhmlau** [comment]
@slnode test please

**Pull request closed without merging**